src/Portal.sol

```diff
@@ -171,23 +172,40 @@ abstract contract Portal is NttManagerNoRateLimiting, IPortal {
         if (recipient_ == bytes32(0)) revert InvalidRecipient();
         if (refundAddress_ == bytes32(0)) revert InvalidRefundAddress();
 
+        IERC20 mToken_ = IERC20(token);
+        uint256 startingBalance_ = mToken_.balanceOf(address(this));
+
         // transfer source token from the sender
         IERC20(sourceToken_).transferFrom(msg.sender, address(this), amount_);
 
         // if the source token isn't M token, unwrap it
-        if (sourceToken_ != token) {
+        if (sourceToken_ != address(mToken_)) {
             IWrappedMTokenLike(sourceToken_).unwrap(address(this), amount_);
         }
 
-        // Burn M tokens on Spoke.
-        // Since SpokePortal is non-earner the exact amount is equal to the actual amount received from the sender.
+        // The actual amount of M tokens that Portal received from the sender.
+        // Accounts for potential rounding errors when transferring between earners and non-earners,
+        // as well as potential fee-on-transfer functionality in the source token.
+        uint256 actualAmount_ = mToken_.balanceOf(address(this)) - startingBalance_;
+
+        if (amount_ > actualAmount_) {
+            unchecked {
+                // If the difference between the specified transfer amount and the actual amount exceeds
+                // the maximum acceptable rounding error (e.g., due to fee-on-transfer in an extension token)
+                // transfer the actual amount, not the specified.
+
+                // Otherwise, the specified amount will be transferred and the deficit caused by rounding error will
+                // be covered from the yield earned by HubPortal.
+                if (amount_ - actualAmount_ > _getMaxRoundingError()) {
+                    amount_ = actualAmount_;
+                }
+            }
+        }
+
+        // Burn the actual amount of M tokens on Spoke.
         // In case of Hub, do nothing, as tokens are already transferred.
-        _burnOrLock(amount_);
+        _burnOrLock(actualAmount_);
```

Comment:
Shouldn't we still burn `amount_` here?

Reply:
plus +1 on this, we should still burn `amount`?